modules/ref_fb_module/src/statistics_fb_impl.cpp

```diff
@@ -52,22 +53,70 @@ void StatisticsFbImpl::initProperties()
     objPtr.getOnPropertyValueWrite("DomainSignalType") +=
         [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { propertyChanged(); };
 
+    objPtr.addProperty(BoolProperty("TriggerMode", false));
+    objPtr.getOnPropertyValueWrite("TriggerMode") +=
+        [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { triggerModeChanged(); };
+
     readProperties();
 }
 
 void StatisticsFbImpl::propertyChanged()
 {
     std::scoped_lock lock(sync);
     readProperties();
-
     configure();
 }
 
+void StatisticsFbImpl::triggerModeChanged()
+{
+    sync.lock();
```

Comment:
Use try/finally statements when locking manually and not using scoped_lock which uses RAII concept to make sure the resource is unlocked. This code has a potential to dead lock the system if exception occurs.

Reply:
I have made  what I believe are the appropriate changes.